spec/controllers/api/v1/marketing/lists_controller_spec.rb

```diff
@@ -0,0 +1,250 @@
+
+require 'rails_helper'
+
+RSpec.describe Api::V1::Marketing::ListsController do
+
+    # 
+    # initialize()
+    # 
+    def initialize (x)
+        @base_earl = 'api/marketing/lists?'
+        @bugger = false
+    end
+
+    # 
+    # show_test()
+    # 
+    def show_test (t, p)
+        puts ''
+        puts '  trying ' + @base_earl + p.to_query
+        puts '    [' + t + ']'
+    end
+
+    # 
+    # show_subtest()
+    # 
+    def show_subtest (message, result = true)
+        printf "    %-80s %s\n", message, (result ? '[OK]' : '*** FAILED ***')
+    end
+
+    # 
+    # bugger_add()
+    # 
+    def bugger_add (message)
+        pp message if @bugger
+    end
+
+    # 
+    # describe
+    # 
+    describe 'GET get_list' do
+
+        # get with no params (code 1, method not allowed)
+        t1 = 'get with no params (code 1, method not allowed)'
+        it t1 do
+            params = {}
+            get 'get_list'
+            show_test t1, params
+            json = JSON.parse(response.body)
+            expect(response).to be_success
+            show_subtest 'response be_success'
+            expect(response).to have_http_status(200)
+            show_subtest 'have_http_status 200'
+            bugger_add json # bugger
+            expect(json['code']).to be == 1
+            show_subtest 'code == ' + json['code'].to_s
+            show_subtest 'message == ' + json['message']
+        end
+
+        # get with bugger and no auth (code 2, auth fail)
+        t2 = 'get with bugger and no auth (code 2, auth fail)'
+        it t2 do
+            params = {
+                'bugger' => 1,
+            }
+            get 'get_list', params
+            show_test t2, params
+            json = JSON.parse(response.body)
+            bugger_add json # bugger
+            expect(json['code']).to be == 2
+            show_subtest 'code == ' + json['code'].to_s
+            show_subtest 'message == ' + json['message']
+            show_subtest 'bugger[2] == ' + json['bugger'][2]
+            expect(json['bugger'][2]).to be == 'ip: 0.0.0.0'
+        end
+
+        # post with user and bad key (code 2, auth fail)
+        t3 = 'post with user and bad key (code 2, auth fail)'
+        it t3 do
+            params = {
+                'key' => 'xxxxxxxxx',
+                'user' => 'emma',
+            }
+            post 'get_list', params
+            show_test t3, params
+            json = JSON.parse(response.body)
+            bugger_add json # bugger
+            expect(json['code']).to be == 2
+            show_subtest 'code == ' + json['code'].to_s
+            show_subtest 'message == ' + json['message']
+        end
+
+        # post with bad user and key (code 2, auth fail)
+        t4 = 'post with bad user and key (code 2, auth fail)'
+        it t4 do
+            params = {
+                'key' => '2b0fde9309034dc07de88a6a52be1bb81a5501a4bb329f3ada929296971b72d5',
```

Comment:
hmm. okay. I suspect this may cause the deployment to preprod to fail, because it probably runs the tests as part of the deployment pipeline. But it;s worth a try. Another way to do this might be to set the expected key (overriding the real key) as part of the test setup.